R/get_uptake_data.r

```diff
@@ -0,0 +1,153 @@
+    stop(paste("There is no column called ", cumulative, sep = ""))
+  }
+
+  if (state %in% colnames(data)) {
+    state_col <- data[, state]
+  } else {
+    state_col <- rep_len(state, nrow(data))
+  }
+
+  if (!is.null(state_key)) {
+    state_key <- read.csv(state_key)
+    state_idx <- match(state_col, state_key[, 2], nomatch = 0) +
+      match(state_col, state_key[, 1], nomatch = 0)
+    state_idx[state_idx == 0] <- NA
+    state_col <- state_key[, 1][state_idx]
+    state_col <- state_col[!is.na(state_col)]
+    data <- data[!is.na(state_idx), ]
+    cuml_col <- cuml_col[!is.na(state_idx)]
+  }
+
+  if (all(date %in% colnames(data))) {
+    date_col <- data[, date]
+    if (length(date) > 1) {
+      for (i in seq_along(date)) {
+        if (any(grepl("^[A-Za-z]", date_col[, i]))) {
+          date_col[, i] <- sub(".*-", "", date_col[, i])
+          date_col[, i] <- trimws(date_col[, i], "both")
+          date_col[, i] <- sub(" ", "/", date_col[, i])
+          date_col[, i] <- stringr::str_replace_all(date_col[, i],
+                                                    "January", "1")
+          date_col[, i] <- stringr::str_replace_all(date_col[, i],
+                                                    "February", "2")
+          date_col[, i] <- stringr::str_replace_all(date_col[, i],
+                                                    "March", "3")
+          date_col[, i] <- stringr::str_replace_all(date_col[, i],
+                                                    "April", "4")
+          date_col[, i] <- stringr::str_replace_all(date_col[, i],
+                                                    "May", "5")
+          date_col[, i] <- stringr::str_replace_all(date_col[, i],
+                                                    "June", "6")
+          date_col[, i] <- stringr::str_replace_all(date_col[, i],
+                                                    "July", "7")
+          date_col[, i] <- stringr::str_replace_all(date_col[, i],
+                                                    "August", "8")
+          date_col[, i] <- stringr::str_replace_all(date_col[, i],
+                                                    "September", "9")
+          date_col[, i] <- stringr::str_replace_all(date_col[, i],
+                                                    "October", "10")
+          date_col[, i] <- stringr::str_replace_all(date_col[, i],
+                                                    "November", "11")
+          date_col[, i] <- stringr::str_replace_all(date_col[, i],
+                                                    "December", "12")
+        }
+      }
+      date_col <- do.call(paste, c(date_col, sep = "/"))
+    }
+  } else {
+    date_col <- rep_len(date, nrow(data))
+  }
+  date_col <- as.Date(date_col, format = date_format)
+
+  elapsed_col <- as.numeric(date_col - as.Date(start_date,
+                                               format = date_format))
+
+  year_col <- as.numeric(format(date_col, "%Y")) -
```

Comment:
is the idea here to get a "season"? Like, 2023/2024 season runs July 2023 to Aug 2024?